CRISPResso2/CRISPRessoShared.py

```diff
@@ -1661,3 +1672,267 @@ def zip_results(results_folder):
+    if aln_rates is not None:
+        messages.append(message)
+    highRateOfSubstitutionsOutsideWindowGuardRail = HighRateOfSubstitutionsOutsideWindowGuardRail(messageHandler, outsideWindowMaxSubRate)
+    rate = highRateOfSubstitutionsOutsideWindowGuardRail.safety(aln_stats['N_GLOBAL_SUBS'], aln_stats['N_SUBS_OUTSIDE_WINDOW'])
+    if rate is not None:
+        messages.append(rate)
+    highRateOfSubstitutions = HighRateOfSubstitutionsGuardRail(messageHandler, maxRateOfSubs)
+    rate = highRateOfSubstitutions.safety(aln_stats['N_MODS_IN_WINDOW'], aln_stats['N_MODS_OUTSIDE_WINDOW'], aln_stats['N_GLOBAL_SUBS'])
+    if rate is not None:
+        messages.append(rate)
+    amplicons = {}
+    guide_groups = []
+    for name in crispresso2_info['results']['ref_names']:
+        amplicons[name] = crispresso2_info['results']['refs'][name]['sequence_length']
+        guide_groups.append(crispresso2_info['results']['refs'][name]['sgRNA_sequences'])
+    unique_guides = {}
+    for guides in guide_groups:
+        for guide in guides:
+            if guide not in unique_guides.keys():
+                unique_guides[guide] = len(guide)
+    shortAmpliconSequence = ShortSequenceGuardRail(messageHandler, amplicon_len, 'amplicon')
+    ampliconLength = shortAmpliconSequence.safety(amplicons)
+    if ampliconLength is not None:
+        for message in ampliconLength:
+            messages.append(message)
+    shortGuideSequence = ShortSequenceGuardRail(messageHandler, guide_len, 'guide')
+    guideLength = shortGuideSequence.safety(unique_guides)
+    if guideLength is not None:
+        for message in guideLength:
+            messages.append(message)
+    longAmpliconShortReadsGuardRail = LongAmpliconShortReadsGuardRail(messageHandler, ampliconToReadLen)
+    ampliconLongerThanRead = longAmpliconShortReadsGuardRail.safety(amplicons, aln_stats['READ_LENGTH'])
+    if ampliconLongerThanRead is not None:
+        for message in ampliconLongerThanRead:
+            messages.append(message)
+    crispresso2_info['results']['guardrails'] = messages
+
+class GuardRailMessageHandler:
+    def __init__(self, logger):
+        self.logger = logger
+
+    def display_warning(self, message):
+        self.logger.warning(message)
+
+    def report_warning(self, message):
+        html_warning = '<div class="alert alert-danger"><strong>Guardrail Warning!</strong>{0}</div>'.format(message)
+        return html_warning
+
+class TotalReadsGuardRail:
+    def __init__(self, messageHandler, minimum):
+        self.messageHandler = messageHandler
+        self.minimum = minimum
+        self.message = " Low number of total reads: <{}".format(minimum)
+
+    def safety(self, total):
+        if total < self.minimum:
+            self.messageHandler.display_warning(self.message)
+            return self.messageHandler.report_warning(self.message)
+        return None
```

Comment:
By default the function will return `None`, but that is fine if you want to be explicit.